src/CommunityToolkit.Maui.Core/Layouts/DockLayoutManager.shared.cs

```diff
@@ -0,0 +1,152 @@
+﻿using CommunityToolkit.Maui.Core.Interfaces;
+using CommunityToolkit.Maui.Core.Primitives;
+using Microsoft.Maui.Layouts;
+
+namespace CommunityToolkit.Maui.Core.Layouts;
+
+/// <summary>LayoutManager for <see cref="IDockLayout"/>.</summary>
+public class DockLayoutManager : LayoutManager, IDockLayoutManager
+{
+	readonly IDockLayout dockLayout;
+
+	/// <summary>Initialize a new instance of <see cref="DockLayoutManager"/>.</summary>
+	public DockLayoutManager(IDockLayout dockLayout)
+		: base(dockLayout)
+	{
+		this.dockLayout = dockLayout;
+	}
+
+	/// <inheritdoc />
+	public override Size Measure(double widthConstraint, double heightConstraint)
+	{
+		var padding = dockLayout.Padding;
+		var horizontalSpacing = dockLayout.HorizontalSpacing;
+		var verticalSpacing = dockLayout.VerticalSpacing;
+
+		var width = padding.HorizontalThickness;
+		var height = padding.VerticalThickness;
+
+		var widthLimit = widthConstraint - padding.HorizontalThickness;
+		var heightLimit = heightConstraint - padding.VerticalThickness;
+
+		foreach (var child in dockLayout)
+		{
+			if (child.Visibility != Visibility.Visible)
+			{
+				continue;
+			}
+
+			var childSize = child.Measure(widthLimit, heightLimit);
+			var position = dockLayout.GetDockPosition(child);
+
+			switch (position)
+			{
+				case DockPosition.Left:
+				case DockPosition.Right:
+					var childWidth = childSize.Width + horizontalSpacing;
+					width += childWidth;
+					widthLimit -= childWidth;
+					break;
+
+				case DockPosition.Top:
+				case DockPosition.Bottom:
+					var childHeight = childSize.Height + verticalSpacing;
+					height += childHeight;
+					heightLimit -= childHeight;
+					break;
+
+				case DockPosition.None:
+					width += childSize.Width;
+					widthLimit -= childSize.Width;
+					height += childSize.Height;
+					heightLimit -= childSize.Height;
+					break;
+
+				default:
```

Comment:
Let's combine None and default, what do you think?

Reply:
I purposely kept them separate. 

As a best practice, when performing a `switch` on an `enum`, it's best to use the `default` case to throw a `NotSupportedException`. 

This ensures that when a new value is added to the `enum` that we don't forget to implement the new value in every `switch`.

Reply:
In that case let’s add it to arrange method as well